### ISSUE TITLE
setting preferred terminal emulator for Vim (Mac OS X)

### ISSUE THREAD

**Issue by nhejazi** [jalvesaq/Nvim-R#190]
I've been using this plugin with Neovim for a while now. Due to various restrictions, I'd like to use it with Vim, which appears to require using `tmux`. After installing `tmux` and following the appropriate instructions, I can use `Nvim-R` from Vim; however, the plugin launches the R session in a terminal emulator different from my preferred one.

Specifically, if launch R using the plugin from within Vim in iTerm, a new R session will start in a new iTerm tab, and the connection between Vim and R will work fine; however, if I launch R using the plugin from within Vim in [Hyper](https://hyper.is/), _a new iTerm window_ will open with the R session, and the connection appears to work fine...I'd like the same behavior of simply opening a new tab, as opposed to opening a new app/window.

It appears that this ought to be controlled by setting `let R_term` or `let R_term_cmd` appropriately in my `vimrc`; however, I'm not sure how to get this working with [Hyper](https://hyper.is/). Any previous issues similar to this?

**Comment by jalvesaq** [comment]
It seems that `R_term` does not work on Mac OS X. Please, read the Nvim-R documentation:
```
:h R_term
```
I guess that you should configure your system to call Hyper as the default terminal emulator, but I don't have any Mac OS X to make tests.

**Comment by nhejazi** [comment]
Thanks for the quick response.

I just had a chance to test this out on my Ubuntu machine, which produces a similar problem. When I start an R session from within Vim (using Hyper as the terminal emulator), a new window of the system's default terminal emulator opens with the requested R session. Again, the connection between the Vim session and R session works great, but it is rather annoying having to deal with two terminal apps when one would suffice. This appears to replicate the issue I noted above on macOS.

To my understanding, there is [no default terminal on macOS](https://superuser.com/questions/379342/setting-iterm2-as-the-default-terminal-osx-lion/379354#379354), so I'm unsure why Nvim-R is calling iTerm rather than Hyper -- I would think that if it were making a call to the default system terminal emulator, the R session would launch in a new terminal.app window. Any advice on how I might be able to resolve this would be appreciated.

For reference, the vimrc I'm using is on GitHub [here](https://github.com/nhejazi/vimForLife/blob/master/vimrc).

**Comment by jalvesaq** [comment]
I don't have Hyper installed, so you have to read Hyper's documentation to know how to call it as explained in the Nvim-R's documentation. Please, do:
```vim
:h R_term_cmd
```

**Comment by nhejazi** [comment]
Thanks -- I've now looked into this and tested out the use of `R_term_cmd`. [This issue](https://github.com/zeit/hyper/issues/462) appears to indicate that launching Hyper from the command line while executing a new app/program __does not work__. Setting `R_term_cmd = "hyper"` on Ubuntu does cause a new instance of Hyper to be launched, but without an R session running.

Closing since this appears to be an issue on Hyper's end. Thanks for the help.
